spring-integration-file/src/main/java/org/springframework/integration/file/FileWritingMessageHandler.java

```diff
@@ -367,6 +374,75 @@ public void setPreserveTimestamp(boolean preserveTimestamp) {
 		this.preserveTimestamp = preserveTimestamp;
 	}
 
+	/**
+	 * String setter for Spring XML convenience.
+	 * @param chmod permissions as an octal string e.g "600";
+	 * @see #setChmod(int)
+	 * @since 5.0
+	 */
+	public void setChmodOctal(String chmod) {
+		Assert.notNull(chmod, "'chmod' cannot be null");
+		setChmod(Integer.parseInt(chmod, 8));
+	}
+
+	/**
+	 * Set the file permissions after uploading, e.g. 0600 for
+	 * owner read/write. Only applies to file systems that support posix
+	 * file permissions.
+	 * @param chmod the permissions.
+	 * @throws IllegalArgumentException if the value is higher than 0777.
+	 * @since 5.0
+	 */
+	public void setChmod(int chmod) {
+		Assert.isTrue(chmod <= 0777, "'chmod' must be between 0 and 0777 (octal)");
```

Comment:
Sure thing! I also going to merge those two warns and move null check to the existing warn message.
Will also run all the tests on Windows to be sure that we haven't missed POSIX hooks.